testserver/testserver.go

```diff
@@ -445,19 +463,25 @@ func NewTestServer(opts ...TestServerOpt) (TestServer, error) {
 
 	nodes := make([]nodeInfo, serverArgs.numNodes)
 	var initArgs []string
+	portArgsStr := make([]string, 3)
+	hostPort := serverArgs.httpPorts[0]
+	for i, port := range serverArgs.httpPorts {
+		portArgsStr[i] = fmt.Sprint(port)
+	}
 	for i := 0; i < serverArgs.numNodes; i++ {
 		nodes[i].state = stateNew
 		nodes[i].listeningURLFile = filepath.Join(baseDir, fmt.Sprintf("listen-url%d", i))
 		if serverArgs.numNodes > 1 {
+			joinArg := fmt.Sprintf("--join=localhost:%s", strings.Join(portArgsStr, ",localhost:"))
 			nodes[i].startCmdArgs = []string{
 				serverArgs.cockroachBinary,
 				startCmd,
 				secureOpt,
 				storeArg + strconv.Itoa(i),
-				fmt.Sprintf("--listen-addr=localhost:%d", 26257+i),
-				fmt.Sprintf("--http-addr=localhost:%d", 8080+i),
+				fmt.Sprintf("--listen-addr=localhost:%d", serverArgs.httpPorts[i]),
+				fmt.Sprintf("--http-addr=localhost:%d", 0),
```

Comment:
for this to be backwards compatible, i think we want it so still default to 26257/8080 if no port is specified

Reply:
I updated it for 26257 listenaddr but I don't think we actually use the HTTP addr in this case, I think it should be okay to just update it use 0 unless we find that we really need to set it.

Reply:
Added a diff parameter to allow people to specify HTTP port as well.